aws/wafv2_helper.go

```diff
@@ -457,20 +568,116 @@ func expandWafv2RuleAction(l []interface{}) *wafv2.RuleAction {
 	action := &wafv2.RuleAction{}
 
 	if v, ok := m["allow"]; ok && len(v.([]interface{})) > 0 {
-		action.Allow = &wafv2.AllowAction{}
+		action.Allow = expandWafv2AllowAction(v.([]interface{}))
 	}
 
 	if v, ok := m["block"]; ok && len(v.([]interface{})) > 0 {
-		action.Block = &wafv2.BlockAction{}
+		action.Block = expandWafv2BlockAction(v.([]interface{}))
 	}
 
 	if v, ok := m["count"]; ok && len(v.([]interface{})) > 0 {
-		action.Count = &wafv2.CountAction{}
+		action.Count = expandWafv2CountAction(v.([]interface{}))
+	}
+
+	return action
+}
+
+func expandWafv2AllowAction(l []interface{}) *wafv2.AllowAction {
+	action := &wafv2.AllowAction{}
+	if len(l) == 0 || l[0] == nil {
+		return action
+	}
+
+	m := l[0].(map[string]interface{})
+	if v, ok := m["custom_request_handling"]; ok && len(v.([]interface{})) > 0 {
+		action.CustomRequestHandling = expandWafv2CustomRequestHandling(v.([]interface{}))
+	}
+
+	return action
```

Comment:
in the empty case, we generally return nil instead of a new API object. we can also directly try to cast the map value by doing the following:
```suggestion
	if len(l) == 0 || l[0] == nil {
		return nil
	}

	m, ok := l[0].(map[string]interface{})
	if !ok { 
	        return nil
        }
        
       action := &wafv2.AllowAction{}

	if v, ok := m["custom_request_handling"].([]interface{}); ok && len(v) > 0 {
		action.CustomRequestHandling = expandWafv2CustomRequestHandling(v)
	}

	return action
```

Reply:
@anGie44 so I actually went ahead and merged the suggestions here, and now the acceptance tests are failing. There is a bit of an odd nuance going on here in that the `allow`, `count` and `block` blocks can be empty. In fact, before my PR, they always were empty. There is a check in the layer above to decide which action is selected. Once we get to `expand` methods, we already know we at least want an empty object. We simply need to decide whether to set any of the additional attributes on that action as well. I'm going to revert these particular changes to get the tests passing again. If you have further suggestions on how to change these in a way that doesn't break  the functionality, I'm open.

Reply:
Totally correct here @andyalm ! Oversight on my part. Yep the wafv2 api is certainly different as it previously accepted the empty structs for allow/count/block  👍